include/nbl/ext/ImGui/ImGui.h

```diff
@@ -1,76 +1,133 @@
 #ifndef _NBL_EXT_IMGUI_UI_H_
 #define _NBL_EXT_IMGUI_UI_H_
 
+#include "nbl/video/declarations.h"
+#include "nbl/asset/IAssetManager.h"
+
 namespace nbl::ext::imgui
 {
 class UI final : public core::IReferenceCounted
 {
 	public:
-		UI(core::smart_refctd_ptr<video::ILogicalDevice> device, int maxFramesInFlight, video::IGPURenderpass* renderpass, video::IGPUPipelineCache* pipelineCache, core::smart_refctd_ptr<ui::IWindow> window);
+		struct MDI
+		{
+			using COMPOSE_T = nbl::video::StreamingTransientDataBufferST<nbl::core::allocator<uint8_t>>;								//! composes memory available for the general purpose allocator to suballocate memory ranges								
+			using SUBALLOCATOR_TRAITS_T = nbl::core::address_allocator_traits<nbl::core::LinearAddressAllocatorST<uint32_t>>;			//! traits for MDI buffer suballocator - fills the data given the mdi allocator memory request
+
+			enum E_BUFFER_CONTENT : uint8_t
+			{
+				EBC_DRAW_INDIRECT_STRUCTURES,
```

Comment:
if you make it an `enum class` then you can skip the `E_` in the name and the `EBC_` prefices in the enums